frontend/lib/resolution/Resolver.cpp

```diff
@@ -460,41 +462,90 @@ types::QualifiedType Resolver::typeErr(const uast::AstNode* ast,
   return t;
 }
 
-static bool isOuterVariable(Resolver* rv, ID target) {
-  if (target.isEmpty()) return false;
+static bool
+isOuterVariable(Resolver& rv, const ID& target, const ID& mention) {
+  if (!target || !mention || target.isSymbolDefiningScope()) return false;
```

Comment:
I would think that the Resolver knows if there is a parent frame going. Basically, I would think that the Resolver knows if it's resolving a nested function. Can't this function have an early return of `false` in the case that the Resolver isn't working on a nested function? (or maybe, in the case that the Resolver isn't working on a function that contains nested functions?)